otx/algorithms/segmentation/adapters/mmseg/tasks/stage.py

```diff
@@ -136,6 +142,17 @@ def configure_classes(self, cfg, task_adapt_op):
         self.org_model_classes = org_model_classes
         self.model_classes = model_classes
 
+    def configure_head(self, cfg):
+        """Change head to custom head to align with otx
+        inteface for class incremental learning."""
+        decode_head = cfg.model.get("decode_head", None)
+        auxiliary_head = cfg.model.get("auxiliary_head", None)
+
+        for head in (decode_head, auxiliary_head):
+            if head:
+                head.head_name = head.type
+                head.type = get_head
```

Comment:
I think this code flow is quite confusing.. I don't think head.type is to be custom function for mmcv-base implementation. I hope we can deal all in OTXCustomHead class.

Reply:
`head.type` can be a function actually. It can be class, function or string (registered head in Registry).
As I  said above - I can create class instead, that will return new class in callback metrhod, but it will be quite similar to current flow

Reply:
Oh no, we can't use Class factory unfortunately. Only function is possible to support inheritance on the go